### PR TITLE
feat/CB2-10024 - Re-introduce Google Tag Manager

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Ticket title
Re-introduces GTM using a new container ID, set up to allow for the values to be stored in git secrets.
_One line description_
[CB2-10024](https://dvsa.atlassian.net/browse/CB2-10024)

## Checklist

- [x] Branch is rebased against the latest develop/common
- [x] Code and UI has been tested manually after the additional changes
- [x] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Delete branch after merge

### CODE REVIEW

File: src/environments/environment.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export const environment = {
   DOCUMENT_RETRIEVAL_API_KEY: '',
   FEEDBACK_URI: '',
   SENTRY_DSN: '',
+  VTM_GTM_CONTAINER_ID: '',
```
Comment (pbardy2000) [inline src/environments/environment.ts:18]: Forgot to mention this, but this also needs adding to environment.prod.ts

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ describe('AppComponent', () => {
   const MockUserService = {
     getUserName$: jest.fn().mockReturnValue(new Observable()),
   };
+  (window as any).gtag = () => {};
```
Comment (naathanbrown) [inline src/app/app.component.spec.ts:17]: Can we suppress this error?

**Review by shivangidas — APPROVED** [review]

File: src/environments/environment.prod.ts [hunk 3]
```diff
@@ -11,4 +11,6 @@ export const environment = {
   DOCUMENT_RETRIEVAL_API_KEY: '',
   FEEDBACK_URI: '',
   SENTRY_DSN: '',
+  VTM_GTM_CONTAINER_ID: '',
+  VTM_GTM_MEASUREMENT_ID: '',
```
Comment (naathanbrown) [inline src/environments/environment.prod.ts:15]: Have you spoke to platform and get this added in there too? Not sure if you are aware of that step

Comment (naathanbrown) [reply src/environments/environment.prod.ts:15]: Needs to be for all non dev environments too

Comment (tomcrawleyy) [reply src/environments/environment.prod.ts:15]: Yeah dont worry there is another task which is in progress for G-Force to add the properties into AWS Secrets

Comment (tomcrawleyy) [reply src/environments/environment.prod.ts:15]: This is for all envs too

File: src/app/app.component.spec.ts (hunk 2 above)
Comment (shivangidas) [reply src/app/app.component.spec.ts:17]: showing a lint error `Unexpected any. Specify a different type`

Comment (tomcrawleyy) [reply src/app/app.component.spec.ts:18]: resolved, thanks

**Review by shivangidas — APPROVED** [review]

**Review by BrandonT95 — APPROVED** [review]

**Pull request merged**